### PR TITLE
Observation refactor to bootstrap-4

### PR DESCRIPTION
**Pull request by mo-nathan**
Merges recent changes on master into bootstrap-4.

### CODE REVIEW

**Comment by mo-nathan** [comment]
This merge looks reasonable to me, but I'd love to have either @JoeCohen or @nimmolo look it over and merge if it looks good to them.

File: app/controllers/vote_controller.rb [hunk 1]
```diff
@@ -23,8 +23,7 @@ def cast_vote
     pass_query_params
     naming = Naming.find(params[:id].to_s)
     observation = naming.observation
-    value = params[:value].to_i
-    observation.change_vote(naming, value)
+    observation.change_vote(naming, params[:value])
     redirect_with_query(controller: :observer,
                         action: :show_observation,
                         id: observation.id)
```
Comment (JoeCohen) [inline app/controllers/vote_controller.rb]: There's a conflict (see below), which should be resolved as follows:

```suggestion
    observation = naming.observation
    observation.change_vote(naming, params[:value])
    redirect_to observation_path(observation.id, q: get_query_param)
```

@mo-nathan: What's the best practice for resolving conflicts for PR's that simply port `master` changes to `bootstrap-4`
Is it:
(a) suggest modifications to the PR?
(b) close the PR, merge `master` into `bootstrap-4`, fix conflicts, create a new PR, merge the new PR?
(c) something else?
```ruby
<<<<<<< HEAD:app/controllers/votes_controller.rb
    @observation = naming.observation
    value = params[:value].to_i
    @observation.change_vote(naming, value)
    # redirect_with_query(
    #   controller: :observations,
    #   action: :show,
    #   id: @observation.id
    # )
    redirect_to observation_path(@observation.id, q: get_query_param)
=======
    observation = naming.observation
    observation.change_vote(naming, params[:value])
    redirect_with_query(controller: :observer,
                        action: :show_observation,
                        id: observation.id)
>>>>>>> master:app/controllers/vote_controller.rb
```

**Comment by JoeCohen** [comment]
1. My apologies if you received multiple notices about my suggested change; I haven't wuite figured out how to use GitHub Review.
2. I'd like to get this merged soon so that it doesn't interfere with `bootstrap-4` development. Because I'll be tied up or away most of the week, I might merge it tonite or tomorrow a.m. if there are no additional comments.

File: app/controllers/vote_controller.rb (hunk 1 above)
Comment (nimmolo) [reply app/controllers/vote_controller.rb]: @JoeCohen Looks like you already did this fix. This looks OK to merge, to me.

Comment (JoeCohen) [reply app/controllers/vote_controller.rb]: I don't think this fix has actually been made, even though GitHub implies that it has.
I probably screwed something up.
I'm going to try again. Sorry if that causes additional emails.

**Pull request merged**